File: TaxiSF/Metadata/DataTransferObjects.cs
namespace Metadata
{
    [Serializable]
    public class LoginDto
    {
        public string Email { get; set; } = "";
        public string Password { get; set; } = "";
    }

    [Serializable]
    public class RegisterDto
    {
        public string Username { get; set; } = "";
        public string Email { get; set; } = "";
        public string Password { get; set; } = "";
        public string FullName { get; set; } = "";
        public DateTime DateOfBirth { get; set; }
        public string Address { get; set; } = "";
        public string Role { get; set; } = "";
        public bool OAuth { get; set; }
        public string ProfileImage { get; set; } = "";
    }

    [Serializable]
    public class UserIdDto
    {
        public int Id { get; set; }
        public string Role { get; set; } = "";
    }

    [Serializable]
    public class UserDto : RegisterDto
    {
        public int Id { get; set; }
        public bool Blocked { get; set; }
        public string AccountStatus { get; set; } = "";
        public float UserRatings { get; set; }
    }

    [Serializable]
    public class AddDriveDto
    {
        public int UserId { get; set; }
        public string StartAddress { get; set; } = "";
        public string EndAddress { get; set; } = "";
        public int Wait { get; set; }
        public float Price { get; set; }
    }

    [Serializable]
    public class DriveDto
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int? DriverId { get; set; }
        public string StartAddress { get; set; } = "";
        public string EndAddress { get; set; } = "";
        public decimal Price { get; set; }
        public int WaitingTime { get; set; }
        public int? TravelTime { get; set; }
        public string DriveStatus { get; set; } = "";
    }

    [Serializable]
    public class DriveState
    {
        public int Drive { get; set; }
        public int UserId { get; set; }
        public int DriverId { get; set; } = -1;
        public int WaitingTime { get; set; }
        public int TravelTime { get; set; } = -1;
        public bool Started { get; set; } = false;
    }
}

File: taxiui/src/Metadata/DataTransferObjects.ts
export interface LoginDto {
  email: string;
  password: string;
}

export interface RegisterDto {
  username: string;
  email: string;
  password: string;
  fullName: string;
  dateOfBirth: Date;
  address: string;
  role: string;
  oauth: boolean;
  profileImage: string;
}

export interface UserIdDto {
  id: number;
  role: string;
}

export interface UserDto extends RegisterDto {
  id: number;
  blocked: boolean;
  accountStatus: string;
  userRatings: number;
}

export interface AddDriveDto {
  userId: number;
  startAddress: string;
  endAddress: string;
  wait: number;
  price: number;
}

export interface DriveDto {
  id: number;
  userId: number;
  driverId?: number | null;
  startAddress: string;
  endAddress: string;
  price: number;
  waitingTime: number;
  travelTime?: number | null;
  driveStatus: string;
}

export interface DriveState {
  drive: number;
  userId: number;
  driverId: number;
  waitingTime: number;
  travelTime?: number | null;
  started: boolean;
}
